kernel/boot_info/src/multiboot2.rs

```diff
@@ -169,6 +190,22 @@ impl crate::BootInformation for multiboot2::BootInformation {
         self.module_tags()
     }
 
+    fn additional_reserved_memory_regions(
+        &self,
+    ) -> Result<Self::AdditionalReservedMemoryRegions, &'static str> {
+        Ok([
+            kernel_memory_region(self)?,
+            bootloader_info_memory_region(self)?,
+            modules_memory_region(self)?,
+        ]
+        .into_iter())
+    }
+
+    fn kernel_end(&self) -> Result<PhysicalAddress, &'static str> {
+        let reserved_region = kernel_memory_region(self)?;
```

Comment:
I don't think so because UEFI only calls it once. `kernel_memory_region` iterates over a slice of approximately ten elements, so either way, I don't think it's worth the effort.